src/core_plugins/kibana/common/field_formats/types/duration.js

```diff
@@ -0,0 +1,74 @@
+import moment from 'moment';
+import { FieldFormat } from '../field_format';
+
+const ratioToSeconds = {
+  picoseconds: 0.000000000001,
+  nanoseconds: 0.000000001,
+  microseconds: 0.000001
+};
+const HUMAN_FRIENDLY = 'humanize';
+const DEFAULT_OUTPUT_PRECISION = 2;
+const DEFAULT_INPUT_FORMAT = { text: 'Seconds', kind: 'seconds' };
+const inputFormats = [
+  { text: 'Picoseconds', kind: 'picoseconds' },
+  { text: 'Nanoseconds', kind: 'nanoseconds' },
+  { text: 'Microseconds', kind: 'microseconds' },
+  { text: 'Milliseconds', kind: 'milliseconds' },
+  DEFAULT_INPUT_FORMAT,
+  { text: 'Minutes', kind: 'minutes' },
+  { text: 'Hours', kind: 'hours' },
+  { text: 'Days', kind: 'days' },
+  { text: 'Weeks', kind: 'weeks' },
+  { text: 'Months', kind: 'months' },
+  { text: 'Years', kind: 'years' }
+];
+const DEFAULT_OUTPUT_FORMAT = { text: 'Human Readable', method: 'humanize' };
+const outputFormats = [
+  DEFAULT_OUTPUT_FORMAT,
+  { text: 'Milliseconds', method: 'asMilliseconds' },
+  { text: 'Seconds', method: 'asSeconds' },
+  { text: 'Minutes', method: 'asMinutes' },
+  { text: 'Hours', method: 'asHours' },
+  { text: 'Days', method: 'asDays' },
+  { text: 'Weeks', method: 'asWeeks' },
+  { text: 'Months', method: 'asMonths' },
+  { text: 'Years', method: 'asYears' }
+];
+
+export class DurationFormat extends FieldFormat {
+  isHuman() {
+    return this.param('outputFormat') === HUMAN_FRIENDLY;
+  }
+  _convert(val) {
+    const inputFormat = this.param('inputFormat');
+    const outputFormat = this.param('outputFormat');
+    const outputPrecision = this.param('outputPrecision');
+    const human = this.isHuman();
+    const prefix = val < 0 && human ? 'minus ' : '';
+    const duration = parseInputAsDuration(val, inputFormat);
+    const formatted = duration[outputFormat]();
+    const precise = human ? formatted : formatted.toFixed(outputPrecision);
+    return prefix + precise;
+  }
+
+  static id = 'duration';
+  static title = 'Duration';
+  static fieldType = 'number';
+
+  static inputFormats = inputFormats;
+  static outputFormats = outputFormats;
+}
+
+DurationFormat.prototype.getParamDefaults = function () {
```

Comment:
Can we move this to a regular method on the `DurationFormat` class instead of modifying the prototype as well?